modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -48,14 +50,36 @@ export class Transaction extends BaseTransaction {
   /** @inheritdoc */
   toJson(): TxData {
     const [acc, time] = this.getTxIdParts();
-    return {
+    const result: TxData = {
       id: acc + '@' + time,
       hash: this.getTxHash(), // TODO: Update once hedera-sdk release this functionality BGA-284
       data: toHex(this._hederaTx.bodyBytes),
       fee: new BigNumber(this._txBody.transactionFee!.toString()).toNumber(),
       from: acc,
       startTime: time,
+      validDuration: new BigNumber(this._txBody.transactionValidDuration!.seconds!.toString()).toNumber(),
+      node: stringifyAccountId(this._txBody.nodeAccountID!),
+      memo: this._txBody.memo,
     };
+
+    if (this._txBody.data === 'cryptoTransfer') {
+      const [recipient, amount] = this.getTransferData();
+      result.amount = amount;
+      result.to = recipient;
+    }
+    return result;
+  }
+
+  private getTransferData(): [string, string] {
+    let transferData;
+    this._txBody.cryptoTransfer!.transfers!.accountAmounts!.forEach(transfer => {
+      const amount = Long.fromValue(transfer.amount!);
```

Comment:
Why exactly are we using `Long`? Can BN suffice here?

Reply:
After talking about it with Thomas we agreed that it's ok for now to use Long. It's the same type used by the protobuf generated type for amount. And using BN would involve some sort of casting to the variable.

Reply:
Yeah, we'll have to get these things out of our system at some point but I don't think this is the time or place